File: Php/src/bowlingGame/BowlingGame.php
<?php

namespace joseahernandez\bowlingGame;

class BowlingGame {
    const FRAMES = 10;
    const MAX_PINS_IN_ROLL = 10;
    const MAX_ROLLS_IN_GAME = 21;

    /**
     * @var int
     */
    private $currentRoll;

    /**
     * @var int[]
     */
    private $rolls;

    /**
     * BowlingGame constructor.
     */
    public function __construct()
    {
        $this->currentRoll = 0;
        $this->rolls = array_fill(0, self::MAX_ROLLS_IN_GAME, 0);
    }

    /**
     * @param int $pins
     */
    public function roll($pins)
    {
        $this->rolls[$this->currentRoll] = $pins;
        $this->currentRoll++;
    }

    /**
     * @return int
     */
    public function score()
    {
        $score = 0;
        $index = 0;

        for ($frame = 0; $frame < self::FRAMES; $frame++) {
            if ($this->isStrike($index)) {
                $score += $this->pointsInStrike($index);
                $index++;
            } else if ($this->isSpare($index)) {
                $score += $this->pointsInSpare($index);
                $index += 2;
            } else {
                $score += $this->rolls[$index] + $this->rolls[$index + 1];
                $index += 2;
            }
        }

        return $score;
    }

    /**
     * @param int $index
     * @return bool
     */
    private function isStrike($index)
    {
        return $this->rolls[$index] === self::MAX_PINS_IN_ROLL;
    }

    /**
     * @param int $index
     * @return int
     */
    private function pointsInStrike($index)
    {
        return self::MAX_PINS_IN_ROLL + $this->rolls[$index + 1] + $this->rolls[$index + 2];
    }

    /**
     * @param int $index
     * @return bool
     */
    private function isSpare($index)
    {
        return $this->rolls[$index] + $this->rolls[$index + 1] === self::MAX_PINS_IN_ROLL;
    }

    /**
     * @param int $index
     * @return int
     */
    private function pointsInSpare($index)
    {
        return self::MAX_PINS_IN_ROLL + $this->rolls[$index + 2];
    }
}

File: Swift/BownlingGameKata/BownlingGameKata/BowlingGame.swift
import Foundation

class BowlingGame {
    private static let FRAMES = 10
    private static let MAX_PINS_IN_ROLL = 10
    
    private var currentRoll: Int
    private var rolls: [Int]
    
    init() {
        currentRoll = 0
        rolls = [Int](count: 21, repeatedValue: 0)
    }
    
    func roll(pins: Int) {
        rolls[currentRoll] = pins
        currentRoll += 1
    }
    
    func score() -> Int {
        var score = 0
        var frame = 0
        for _ in 1...BowlingGame.FRAMES {
            
            switch frame {
                case let fr where isStrike(frame):
                    score += pointsInStrike(fr)
                    frame += 1
                case let fr where isSpare(frame):
                    score += pointsInSpare(fr)
                    frame += 2
                case let fr:
                    score += rolls[fr] + rolls[fr + 1]
                    frame += 2
            }
        }
        
        return score
    }
    
    private func isStrike(frame: Int) -> Bool {
        return rolls[frame] == BowlingGame.MAX_PINS_IN_ROLL
    }
    
    private func pointsInStrike(frame: Int) -> Int {
        return 10 + rolls[frame + 1] + rolls[frame + 2]
    }
    
    private func isSpare(frame: Int) -> Bool {
        return rolls[frame] + rolls[frame + 1] == BowlingGame.MAX_PINS_IN_ROLL
    }
    
    private func pointsInSpare(frame: Int) -> Int {
        return 10 + rolls[frame + 2]
    }
    
}